cedar-lean/Cedar/Validation/RequestEntityValidator.lean

```diff
@@ -0,0 +1,157 @@
+import Cedar.Validation.Validator
+import Cedar.Validation.Typechecker
+namespace Cedar.Validation
+
+open Cedar.Spec
+open Cedar.Data
+
+inductive RequestValidationError where
+| typeError (loc : String)
+
+abbrev RequestValidationResult := Except RequestValidationError Unit
+
+inductive EntityValidationError where
+| typeError (loc : String)
+
+abbrev EntityValidationResult := Except EntityValidationError Unit
+
+def instanceOfBoolType (b : Bool) (bty : BoolType) : Bool :=
+match b, bty with
```

Comment:
Nit: we indent the function body (`match`) and leave `match` cases at the same level of indentation as the `match`.